src/EventTickets/Actions/ConvertEventTicketsBlockToFieldsApi.php

```diff
@@ -34,11 +38,12 @@ public function __invoke(BlockModel $block, int $formId)
                     ->ticketTypes($ticketTypes);
 
                 $eventTicketsField->scope(function (EventTickets $field, $value, Donation $donation) {
-                    if (empty($value)) {
-                        return;
-                    }
 
-                    // (new UpdateDonationWithEventTickets())($value, $donation, $field);
```

Comment:
Besides generating the tickets here, I think we also need to update the donation amount to include the amount paid in tickets like in this draft, similar to what we do in Fee Recovery, otherwise, the donation amount will be shown with an total amount smaller than what have been paid:

```php
$donation->amount = $donation->amount->add(new Money(
    array_sum(array_column(json_decode($value), 'amount')),
    $donation->amount->getCurrency()
));

$donation->save();
```

Reply:
It seems potentially dangerous to update a Donations amount. Should the full amount not be set before the donation is created?

Reply:
Does the field scope process before the payment method is charged the donation? In that case, ok, we will need to update the donation amount.

Does that cause any issue with payment intents?

Reply:
Hm... If the scope runs before the payment gateway is charged, then we'll need to delay the ticket generation until after the charge.

Reply:
Yeah, the scope is more concerned about preparing the donation for being processed (paid). Btw, I think I like the idea of relying on donation status change hooks to create and delete tickets.

Reply:
We'll need to create the tickets here, because this is where the form data is available.

The status update hook won't have that unless we store it somewhere.

@pauloiankoski how about we merge this and consider how to support a ticket status in a new PR? Jason has suggested using the donation status as the ticket status and I think that could work.